### PR TITLE
[Merged by Bors] - TO-3242 integration tests for feed

### PR DESCRIPTION
**Pull request by rustonaut**
**Based on:**

- #556

### CODE REVIEW

File: discovery_engine/test/feed_manager_test.dart [hunk 1]
```diff
@@ -60,145 +60,149 @@ Future<void> main() async {
 
   EventHandler.registerHiveAdapters();
 
-  group('FeedManager', () {
```
Comment (rustonaut) [inline discovery_engine/test/feed_manager_test.dart:63]: moved/turned into storage independent integration tests

File: discovery_engine/test/feed_manager_test.dart [hunk 2]
```diff
@@ -60,145 +60,149 @@ Future<void> main() async {
...
-      expect(
-        feed[2].documentId,
-        equals(engine.feedDocuments[0].document.documentId),
-      );
-      // doc3 is excluded since it is inactive
-    });
-    //TODO[pmk] test use mock engine, check which tests can be recycled
-    //ignore:require_trailing_commas
-  }, skip: cfgFeatureStorage);
+  group(
```
Comment (rustonaut) [inline discovery_engine/test/feed_manager_test.dart:63]: only indention changed due to how `skip:` + `require_trailing_comas` interacts badly

File: discovery_engine/test/integration/feed_test.dart [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      await engine.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('close documents', () async {
+      final batch = expectEvent<NextFeedBatchRequestSucceeded>(
+        await engine.requestNextFeedBatch(),
+      ).items;
+      //FIXME once we have mock data producing multiple batches close muiltiple but not all documents
```
Comment (rustonaut) [inline discovery_engine/test/integration/feed_test.dart:54]: :pushpin: We should extend the mock data to have multiple batches and more then just one document per batch. But that is outside of scope.

Comment (finiteprods) [reply discovery_engine/test/integration/feed_test.dart:54]: :art: for clarity, can you extend this `FIXME` comment to also later uncomment the other assertions below? just to make it clear those assertions relate to this fix

**Review by finiteprods — APPROVED** [review]
one minor comment otherwise lgtm

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**